repositories/AnswerRepository.js

```diff
@@ -55,30 +63,72 @@ const getById = id =>
     .where({ isDeleted: false })
     .then(fromDb);
 
-const insert = ({
-  description,
-  guidance,
-  label,
-  secondaryLabel,
-  qCode,
-  type,
-  mandatory,
-  properties,
-  questionPageId
-}) =>
-  Answer.create(
-    toDb({
-      description,
-      guidance,
-      label,
-      secondaryLabel,
-      qCode,
-      type,
-      mandatory,
-      properties,
-      questionPageId
-    })
-  )
+const createAnswer = async (args, ctx) => {
+  return db.transaction(async trx => {
+    const defaultProperties = getDefaultAnswerProperties(args.input.type);
+    const input = merge({}, args.input, { properties: defaultProperties });
+    const answer = await ctx.repositories.Answer.insert(input, trx);
+
+    if (includes(answer.type, flatten(values(answerTypeMap)))) {
```

Comment:
Going forward I don't think the flatten values approach will work.

Right now you have:

```js
{
  number: ["Number", "Currency"]
}
```

I think this will need to be flipped to something like this:

```js
{
 Number: "number",
 Currency: "number"
}
```

Then you look up which set of validation rules apply with `answerTypeMap[answer.type]`.

Happy to leave this as is for now, because it works in the case where we have a single validation type. But this will need to change when we add date and duration validations.

Reply:
This values flatten only checks that the answer type is a valid type for validations so it should create an array of all accepted answer types. Then if the current answer type is included in those it goes to the create default validations where it uses a couple different lodash functions to work out what validations to put in.

I could be wrong but I *think* this approach should still work with new validations, but it might be too complex just for the sake of keeping the default object a tiny bit smaller.